src/Umbraco.Abstractions/UmbracoContextReference.cs

```diff
@@ -51,7 +49,7 @@ public void Dispose()
             if (IsRoot)
             {
                 UmbracoContext.Dispose();
-                _umbracoContextAccessor.UmbracoContext = null;
+                UmbracoContext = null;
```

Comment:
I think this also changes the logic of how this works, when the reference is disposed it is supposed to clear the context on the accessor. See the code comments on the class: 

```
/// <summary>
/// Represents a reference to an <see cref="UmbracoContext"/> instance.
/// </summary>
/// <remarks>
/// <para>A reference points to an <see cref="UmbracoContext"/> and it may own it (when it
/// is a root reference) or just reference it. A reference must be disposed after it has
/// been used. Disposing does nothing if the reference is not a root reference. Otherwise,
/// it disposes the <see cref="UmbracoContext"/> and clears the
/// <see cref="IUmbracoContextAccessor"/>.</para>
/// </remarks>
```

I'll revert